src/Stratis.Bitcoin/Consensus/Rules/ConsensusRules.cs

```diff
@@ -85,30 +83,30 @@ protected ConsensusRules(Network network, ILoggerFactory loggerFactory, IDateTim
             Guard.NotNull(checkpoints, nameof(checkpoints));
 
             this.Network = network;
-            this.DateTimeProvider = dateTimeProvider;
+
             this.Chain = chain;
-            this.NodeDeployments = nodeDeployments;
-            this.loggerFactory = loggerFactory;
-            this.ConsensusSettings = consensusSettings;
             this.Checkpoints = checkpoints;
             this.ConsensusParams = this.Network.Consensus;
+            this.ConsensusSettings = consensusSettings;
+            this.DateTimeProvider = dateTimeProvider;
+            this.loggerFactory = loggerFactory;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.NodeDeployments = nodeDeployments;
             this.PerformanceCounter = new ConsensusPerformanceCounter(this.DateTimeProvider);
 
             this.consensusRules = new Dictionary<string, ConsensusRule>();
             this.partialValidationRules = new List<ConsensusRuleDescriptor>();
             this.headerValidationRules = new List<ConsensusRuleDescriptor>();
             this.fullValidationRules = new List<ConsensusRuleDescriptor>();
             this.integrityValidationRules = new List<ConsensusRuleDescriptor>();
-
         }
 
         /// <inheritdoc />
-        public ConsensusRules Register(IRuleRegistration ruleRegistration)
+        public ConsensusRules Register()
         {
-            Guard.NotNull(ruleRegistration, nameof(ruleRegistration));
+            Guard.Assert(this.Network.Consensus.Rules.Any());
```

Comment:
Is there a possibility of someone wishing to run a node with no rules whatsoever?